text/0486-deprecate-mouseenter.md

```diff
@@ -0,0 +1,162 @@
+these extremely high-volume events.
+
+While efforts to make this more "pay as you go" are [possible](https://github.com/emberjs/ember.js/pull/17911), 
+the trade-off of keeping support around still seems unfavorable, as these events fire so
+frequently, while they are (most certainly) very rarely used.
+
+This is even more so given that Glimmer Components with their outerHTML semantics do not 
+work with event handler methods, and `{{action}}` will eventually fade away in favor of
+`{{on}}` using native `addListener()`.
+
+## Transition Path
+
+When a `mouseenter`/`mouseleave`/`mousemove` event is dispatched by the EventDispatcher to a component
+or `{{action}}` listening to that event type, a deprecation warning will be issued.
+
+The linked migration guide will cover these use cases:
+
+#### Ember.Component
+
+Before:
+
+```js
+import Component from '@ember/component';
+
+export default class MyComponent extends Component {
+  mouseEnter(e) {
+    // do something
+  }
+}
+```
+
+After:
+
+```js
+import Component from '@ember/component';
+import { action } from '@ember/object';
+
+export default class MyComponent extends Component {
+  @action
+  handleMouseEnter(e) {
+    // do something
+  }
+  
+  didInsertElement() {
+    super.didInsertElement(...arguments);
+    this.element.addEventListener('mouseenter', this.handleMouseEnter);
+  }
+  
+  willDestroyElement() {
+    super.willDestroyElement(...arguments);
+    this.element.removeEventListener('mouseenter', this.handleMouseEnter);
+  }
+}
+```
+
+An alternative to attaching the event listener in the component class is to opt into outer HTML semantics by making the
+component tag-less and using the `{{on}}` modifier in the template:
+
+```js
+import Component from '@ember/component';
+import { action } from '@ember/object';
+
+export default class MyComponent extends Component {
+  tagName = '';
+  
+  @action
+  handleMouseEnter(e) {
+    // do something
+  }
+}
+```
+
+```hbs
+<div {{on "mouseenter" this.handleMouseEnter}}>
+  ...
+</div>
+```
+
+#### `{{action}}` modifier
+
+Before:
+
+```hbs
+<button {{action "handleMouseEnter" on="mouseEnter"}}>Hover</button>
+```
+
+After (based on [RFC471](https://github.com/emberjs/rfcs/blob/master/text/0471-on-modifier.md)):
+
+```hbs
+<button {{on "mouseenter" this.handleMouseEnter}}>Hover</button>
+```
+
+## How We Teach This
+
+The deprecation guide should explain the transition path as shown above.
+
+The Guide's [Handling Events](https://guides.emberjs.com/release/components/handling-events/#toc_event-names)
+section must remove the reference to `mouseenter`, `mouseleave` and `mousemove`.
+
+Other than that, no changes are required, as the replacement APIs are all available and
+well established, and this RFC does not introduce anything new. Also having certain event
+types not be supported by the EventDispatcher is not something new, as other high-volume
+events like `mouseover` or `scroll` are also not supported (by default).
+
```

Comment:
Some additional details:

```suggestion
There will also be one deprecation entry for each event in the [deprecations guide](https://github.com/ember-learn/deprecation-app).

```

Reply:
@jenweber You think there should be separate `.md` content files, i.e. each deprecated event has its own anchor link? And so also different deprecation IDs (in the `deprecate()` call)? 

No problem for me, but why do you think this is useful? The migration examples will be quite repetitive then...

Reply:
I might have misunderstood something from the last framework team meeting. It was described to me that they would be separate depreciations, but if you think it makes sense to combine them, that’s fine by me.

Reply:
Maybe separate deprecation entries for `Ember.Component` vs. `{{action}}`? That would make sense IMHO!